cmd/podman/machine/init.go

```diff
@@ -183,15 +182,21 @@ func initMachine(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("invalid username %q: %w", initOpts.Username, ldefine.RegexError)
 	}
 
+	// the same name and error
 	// Check if machine already exists
-	_, exists, err := shim.VMExists(initOpts.Name, []vmconfigs.VMProvider{provider})
-	if err != nil {
+	var errNotExists *define.ErrVMDoesNotExist
+	_, _, err := shim.VMExists(initOpts.Name)
+	if err != nil && !errors.As(err, &errNotExists) {
```

Comment:
Could we just do 
```suggestion
	if !errors.As(err, &errNotExists) {
```

Looking at the source for `errors.As` it will check if `err == nil` for us https://cs.opensource.google/go/go/+/refs/tags/go1.25.3:src/errors/wrap.go;l=97